src/families/algorand/specs.js

```diff
@@ -0,0 +1,237 @@
+// @flow
+import expect from "expect";
+import invariant from "invariant";
+import type { Transaction } from "./types";
+import { getCryptoCurrencyById, parseCurrencyUnit } from "../../currencies";
+import { isAccountEmpty } from "../../account";
+import { pickSiblings } from "../../bot/specs";
+import type { AppSpec } from "../../bot/types";
+
+const currency = getCryptoCurrencyById("algorand");
+
+// Minimum fees
+const minFees = parseCurrencyUnit(currency.units[0], "0.001");
+
+// Minimum balance required for a new non-ASA account
+const minBalanceNewAccount = parseCurrencyUnit(currency.units[0], "0.1");
+
+// Minimum balance for a non-ASA account
+let getMinBalance = (account) => {
+  const minBalance = parseCurrencyUnit(currency.units[0], "0.1");
+  const numberOfTokens = account.subAccounts
+    ? account.subAccounts.filter((a) => a.type === "TokenAccount").length
+    : 0;
+  return minBalance.multipliedBy(1 + numberOfTokens);
+};
+
+// Spendable balance for a non-ASA account
+let getSpendableBalance = (maxSpendable) => {
+  maxSpendable = maxSpendable.minus(minFees);
+  invariant(maxSpendable.gt(0), "spendable balance is too low");
+  return maxSpendable;
+};
+
+// Ensure that, when the recipient corresponds to an empty account,
+// the amount to send is greater or equal to the required minimum
+// balance for such a recipient
+let checkSendableToEmptyAccount = (amount, recipient) => {
+  if (isAccountEmpty(recipient) && amount.lte(minBalanceNewAccount)) {
+    invariant(
+      amount.gt(minBalanceNewAccount),
+      "not enough funds to send to new account"
+    );
+  }
+};
+
+// Extract asset ID from asset-type subaccount
+let extractAssetId = (subaccount) => {
+  if (subaccount.type !== "TokenAccount") {
+    return null;
+  }
+
+  const assetIdRegex = /algorand\/asa\/(\d+)/g; // e.g, 'algorand/asa/312769'
```

Comment:
this is yet another logic to extract out the "id". this really must be factorized into a function